nixops_gcp/nix/gce.nix

```diff
@@ -41,6 +41,46 @@ let
     '';
   };
 
+  imageOptions = { config, ...}: {
+    options = {
+      name = mkOption {
+        default = null;
+        example = "nixos-18091228a4c4cbb613c-x86-64-linux";
+        type = types.nullOr (types.either types.str (resource "gce-image"));
+        description = ''
+          Name of an existent image or image-resource to be used.
+          Must specify the project if the image is defined as public.
+        '';
+      };
+      family = mkOption {
+        default = null;
+        example = "nixos-20-03";
+        type = types.nullOr types.str;
+        description = ''
+          Image family to grab the latest non-deprecated image from.
+          Must specify the project if the image family is defined as public.
+        '';
+      };
+      project = mkOption {
+        default = null;
+        example = "gcp-project";
+        type = types.nullOr types.str;
+        description = ''
+          The parent project containing a GCE image that was made public
+          for all authenticated users.
+        '';
+      };
+    };
+    # ToDo : These assertions are not working for some reason..
```

Comment:
By the way, @AmineChikhaoui  do you happen to know why these assertions are not working?